### PR TITLE
Replace flake8 and isort with ruff

### PR DESCRIPTION
**Pull request by samuelhwilliams**

### CODE REVIEW

File: requirements_for_test.txt [hunk 1]
```diff
@@ -1,9 +1,8 @@
 -r requirements.txt
 black==22.8.0  # Also update `.pre-commit-config.yaml` if this changes
-flake8==5.0.4  # Also update `.pre-commit-config.yaml` if this changes
-flake8-bugbear==22.9.23
+ruff==0.0.256  # Also update `.pre-commit-config.yaml` if this changes
```
Comment (leohemsted) [inline requirements_for_test.txt:3]: any reason you added 0.0.256 vs a more recent version? (i notice that they’re up to 261)

File: Makefile [hunk 2]
```diff
@@ -82,9 +82,7 @@ generate-version-file: ## Generates the app version file
 
 .PHONY: test
 test: ## Run tests
-	flake8 .
-	isort --check-only ./app ./tests
-	black --check .
```
Comment (leohemsted) [inline Makefile:87]: did we intend to remove `black` here?

**Review by leohemsted — APPROVED** [review]

File: requirements_for_test.txt (hunk 1 above)
Comment (samuelhwilliams) [reply requirements_for_test.txt:3]: 🥲 .256 is so much nicer

But no, I think 256 was the latest version when I made the branch a few weeks ago and hadn't bumped it. But will now.

File: Makefile (hunk 2 above)
Comment (samuelhwilliams) [reply Makefile:87]: No :) (I initially thought ruff could replace black but it doesn't - [yet](https://github.com/charliermarsh/ruff/issues/1904))

**Pull request merged**